optimum/intel/openvino/modeling_diffusion.py

```diff
@@ -0,0 +1,462 @@
+        compress_model_transformation(self.text_encoder.model)
+        compress_model_transformation(self.unet.model)
+        self.clear_requests()
+        return self
+
+    def clear_requests(self):
+        self.text_encoder.request = None
+        self.vae_decoder.request = None
+        self.unet.request = None
+
+    def compile(self):
+        self.text_encoder._create_inference_request()
+        self.vae_decoder._create_inference_request()
+        self.unet._create_inference_request()
+
+    def __call__(self, *args, **kwargs):
+        guidance_scale = kwargs.get("guidance_scale", None)
+        if guidance_scale is not None and guidance_scale <= 1 and not self.is_dynamic:
+            raise ValueError(
+                f"`guidance_scale` was set to {guidance_scale}, static shapes are only supported for `guidance_scale` > 1, "
+                "please set `dynamic_shapes` to `True` when loading the model."
+            )
+        return StableDiffusionPipelineMixin.__call__(self, *args, **kwargs)
+
+    @classmethod
+    def _load_config(cls, config_name_or_path: Union[str, os.PathLike], **kwargs):
+        return cls.load_config(config_name_or_path, **kwargs)
+
+    def _save_config(self, save_directory):
+        self.save_config(save_directory)
+
+
+class OVModelPart:
+    def __init__(
+        self, model: openvino.runtime.Model, parent_model: OVBaseModel, ov_config: Optional[Dict[str, str]] = None
+    ):
+        self.model = model
+        self.parent_model = parent_model
+        self.input_names = {key.get_any_name(): idx for idx, key in enumerate(self.model.inputs)}
+        self.input_dtype = {
+            inputs.get_any_name(): OV_TO_NP_TYPE[inputs.get_element_type().get_type_name()]
+            for inputs in self.model.inputs
+        }
+        self.ov_config = ov_config or self.parent_model.ov_config
+        self.request = None
+
+    def _create_inference_request(self):
+        if self.request is None:
+            logger.info("Compiling the encoder and creating the inference request ...")
+            compiled_model = core.compile_model(self.model, self.device, self.ov_config)
+            self.request = compiled_model.create_infer_request()
+
+    @property
+    def device(self):
+        return self.parent_model._device
+
+
+class OVModelTextEncoder(OVModelPart):
+    def __call__(self, input_ids: np.ndarray):
+
+        self._create_inference_request()
+
+        inputs = {
+            "input_ids": input_ids,
+        }
+        outputs = self.request.infer(inputs)
+        return list(outputs.values())
+
+
+class OVModelUnet(OVModelPart):
+    def __call__(self, sample: np.ndarray, timestep: np.ndarray, encoder_hidden_states: np.ndarray):
+
+        self._create_inference_request()
+
+        inputs = {
+            "sample": sample,
+            "timestep": timestep,
+            "encoder_hidden_states": encoder_hidden_states,
+        }
+
+        outputs = self.request.infer(inputs)
```

Comment:
This was accelerated in the latest version of OpenVINO via non-copy use of passed numpy data. Hope, it will result in some speedup when moving from openvino==2022.3 to openvino=2023.0.*.